kubemarine/plugins/calico.py

```diff
@@ -37,410 +35,243 @@ def enrich_inventory(inventory, cluster):
-                    raise Exception(f"Cannot find original Calico manifest {calico_original_yaml_path}")
-
     return inventory
 
 
-def apply_calico_yaml(cluster, calico_original_yaml, calico_yaml):
+# DEPRECATED
+def apply_calico_yaml(cluster: KubernetesCluster, calico_original_yaml: str, calico_yaml: str):
     """
-    The method implements full proccessing for Calico plugin
+    The method implements full processing for Calico plugin
     :param calico_original_yaml: path to original Calico manifest
     :param calico_yaml: file name of the resulting Calico manifest
     :param cluster: Cluster object
     """
 
     calico_yaml = os.path.basename(calico_yaml)
-    destination = '/etc/kubernetes/%s' % calico_yaml
-
-    # create config for plugin module
-    config = {
-        "source": calico_original_yaml,
-        "destination": destination,
-        "do_render": False
-    }
-
-    # get original YAML and parse it into dict of objects
-    calico_original_yaml_path, _ = plugins.get_source_absolute_pattern(config)
-    obj_list = load_multiple_yaml(calico_original_yaml_path, cluster)
-
-    validate_original(cluster, obj_list)
-
-    patched_list = []
-    excluded_list = []
-
-    # list of objects for enrichment
-    objects_for_enrichment = [
+    processor = CalicoManifestProcessor(cluster, cluster.inventory,
+                                        original_yaml_path=calico_original_yaml,
+                                        destination_name=calico_yaml)
+    processor.apply()
+
+
+class CalicoManifestProcessor(Processor):
+    def __init__(self, cluster: KubernetesCluster, inventory: dict,
+                 original_yaml_path: Optional[str] = None, destination_name: Optional[str] = None):
+        plugin_name = 'calico'
+        version = inventory['plugins'][plugin_name]['version']
+        if original_yaml_path is None:
+            original_yaml_path = f"plugins/yaml/calico-{version}.yaml.original"
+        if destination_name is None:
+            destination_name = f"calico-{version}.yaml"
+        super().__init__(cluster, inventory, plugin_name, original_yaml_path, destination_name)
+
+    def exclude_typha_objects_if_disabled(self, manifest: Manifest) -> None:
+        # enrich 'calico-typha' objects only if it's enabled in 'cluster.yaml'
+        # in other case those objects must be excluded
+        str_value = utils.true_or_false(self.inventory['plugins']['calico']['typha']['enabled'])
+        if str_value == 'false':
+            for key in ("Deployment_calico-typha", "Service_calico-typha", "PodDisruptionBudget_calico-typha"):
+                manifest.exclude(key)
+        elif str_value == 'true':
+            return
+        else:
+            raise Exception(f"plugins.calico.typha.enabled must be set in 'True' or 'False' "
+                            f"as string or boolean value")
+
+    def enrich_configmap_calico_config(self, manifest: Manifest) -> None:
```

Comment:
GitHub shows not friendly diff for some reason. To review the changes, I suggest to open in PyCharm and Git -> Current File -> Compare with Branch... -> main